domain/storage/types.go

```diff
@@ -20,13 +20,27 @@ const (
 // Attrs defines storage attributes.
 type Attrs map[string]string
 
+// StoragePoolOrigin defines the origin value of a storage pool.
+// It can be "user", "built-in", or "provider-default".
+type StoragePoolOrigin string
+
+const (
+	// StoragePoolOriginUser defines a storage pool created by a user.
+	StoragePoolOriginUser StoragePoolOrigin = "user"
```

Comment:
We want to use the int based enums where we can. I would be swapping these over to int based enums that run through the usual tests we have to make sure that they're inline with the DB.